src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,150 @@
+            //     ;
+            //
+            // statement_expression
+            //     : invocation_expression
+            //     | null_conditional_invocation_expression
+            //     | object_creation_expression
+            //     | assignment
+            //     | post_increment_expression
+            //     | post_decrement_expression
+            //     | pre_increment_expression
+            //     | pre_decrement_expression
+            //     | await_expression
+            //     ;
+            var isNullConditionalInvocationExpression = IsNullConditionalInvocationExpression(expressionNode);
+
+            return expressionNode.IsKind(SyntaxKind.InvocationExpression)
+                   || isNullConditionalInvocationExpression
+                   || expressionNode.IsKind(SyntaxKind.ObjectCreationExpression)
+                   || expressionNode is AssignmentExpressionSyntax
+                   || expressionNode.IsKind(SyntaxKind.PreIncrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.PreDecrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.PostIncrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.PostDecrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.AwaitExpression);
+        }
+
+        protected override bool CanBeReplacedByThrowExpression(SyntaxNode syntaxNode)
+        {
+            // C# Throw Expression definition from language reference:
+            // 'A throw expression is permitted in only the following syntactic contexts:
+            // As the second or third operand of a ternary conditional operator ?:
+            // As the second operand of a null coalescing operator ??
+            // As the body of an expression-bodied lambda or method.'
+            var parent = syntaxNode.Parent;
+            if (parent is ConditionalExpressionSyntax conditionalExpressionSyntax)
+            {
+                return syntaxNode.Equals(conditionalExpressionSyntax.WhenTrue)
+                    || syntaxNode.Equals(conditionalExpressionSyntax.WhenFalse);
+            }
+
+            if (parent is BinaryExpressionSyntax binaryExpressionSyntax && binaryExpressionSyntax.IsKind(SyntaxKind.CoalesceExpression))
+            {
+                return syntaxNode.Equals(binaryExpressionSyntax.Right);
+            }
+
+            if (parent is ArrowExpressionClauseSyntax)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsNullConditionalInvocationExpression(ExpressionSyntax expressionSyntax)
+        {
+            if (expressionSyntax is ConditionalAccessExpressionSyntax conditionalAccessExpressionSyntax)
+            {
+                var whenNotNull = conditionalAccessExpressionSyntax.WhenNotNull;
+                return whenNotNull.IsKind(SyntaxKind.InvocationExpression) || IsNullConditionalInvocationExpression(whenNotNull);
+            }
```

Comment:
nit: this is worth a comment.

Reply:
do you have test for

`a?.b()` *and*
`a?.b.c()`

Reply:
Yes, this one https://github.com/Cosifne/roslyn/blob/8c55b2b1b2586aa599860edde720889519ea40aa/src/EditorFeatures/CSharpTest/InlineMethod/CSharpInlineMethodTests.cs#L2408

Reply:
Tag this the potential test case that is needed.

Reply:
Tests are added.
Looking into 
The first case is
ConditionalExpression -> InvocationExpression -> MemberBindingExpression
The second case is
ConditionalExpression -> InvocationExpression -> MemberAccessExpression